src/portal/src/app/project/tag-retention/retention.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+export class Retention {
+    algorithm: string;
+    rules: Array<Rule>;
+    trigger: {
+        kind: string;
+    };
+    scope: {
+        level: string,
+        ref: number;
+    };
+    cap: number;
+    constructor () {
+      this.rules = [];
+      this.algorithm = "or";
+      this.trigger = {
+          kind: ""
+      };
+    }
+}
+export class Rule {
+    isDisabled: boolean;
+    id: number;
+    priority: number;
+    action: string;
+    template: string;
+    params: {
+        num: number
+    };
+    tag_selectors: Array<{kind: string; decoration: string; pattern: string}>;
```

Comment:
Could you please define a class for: {kind: string; decoration: string; pattern: string}. Then you can use that class here.

Reply:
modified